src/Tasks/ResolveAssemblyReferences/Services/ResolveAssemblyReferenceComparer.cs

```diff
@@ -0,0 +1,187 @@
+                   AreStringListsEqual(x.FullTargetFrameworkSubsetNames, y.FullTargetFrameworkSubsetNames) &&
+                   AreTaskItemListsEqual(x.InstalledAssemblySubsetTables, y.InstalledAssemblySubsetTables) &&
+                   AreTaskItemListsEqual(x.InstalledAssemblyTables, y.InstalledAssemblyTables) &&
+                   AreStringListsEqual(x.LatestTargetFrameworkDirectories, y.LatestTargetFrameworkDirectories) &&
+                   AreTaskItemListsEqual(x.ResolvedSDKReferences, y.ResolvedSDKReferences) &&
+                   AreStringListsEqual(x.SearchPaths, y.SearchPaths) &&
+                   AreStringListsEqual(x.TargetFrameworkDirectories, y.TargetFrameworkDirectories) &&
+                   AreStringListsEqual(x.TargetFrameworkSubsets, y.TargetFrameworkSubsets);
+        }
+
+        internal static bool CompareOutput(ResolveAssemblyReferenceResponse x, ResolveAssemblyReferenceResponse y)
+        {
+            if (x == y)
+            {
+                return true;
+            }
+
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return x.DependsOnNETStandard == y.DependsOnNETStandard &&
+                   x.DependsOnSystemRuntime == y.DependsOnSystemRuntime &&
+                   AreTaskItemListsEqual(x.CopyLocalFiles, y.CopyLocalFiles) &&
+                   AreTaskItemListsEqual(x.FilesWritten, y.FilesWritten) &&
+                   AreTaskItemListsEqual(x.RelatedFiles, y.RelatedFiles) &&
+                   AreTaskItemListsEqual(x.ResolvedDependencyFiles, y.ResolvedDependencyFiles) &&
+                   AreTaskItemListsEqual(x.ResolvedFiles, y.ResolvedFiles) &&
+                   AreTaskItemListsEqual(x.SatelliteFiles, y.SatelliteFiles) &&
+                   AreTaskItemListsEqual(x.ScatterFiles, y.ScatterFiles) &&
+                   AreTaskItemListsEqual(x.SerializationAssemblyFiles, y.SerializationAssemblyFiles) &&
+                   AreTaskItemListsEqual(x.SuggestedRedirects, y.SuggestedRedirects); 
+            //&&
+            //       AreTaskItemListsEqual(x.Assemblies, y.Assemblies) &&
+            //       AreTaskItemListsEqual(x.AssemblyFiles, y.AssemblyFiles);
+        }
+
+        private static bool AreStringListsEqual(string[] x, string[] y)
+        {
+            if (x == y)
+            {
+                return true;
+            }
+
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            if (x.Length != y.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < x.Length; i++)
+            {
+                if (x[i] != y[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool AreTaskItemListsEqual(ReadOnlyTaskItem[] x, ReadOnlyTaskItem[] y)
```

Comment:
Same here: This could be implemented with `Enumerable.SequenceEqual()` with a custom equality comparer.

Reply:
Same as above + additional work with custom equality comparer